lib/lz4.c

```diff
@@ -978,17 +978,17 @@ LZ4_FORCE_INLINE int LZ4_compress_generic_validated(
                         matchIndex += dictDelta;   /* make dictCtx index comparable with current context */
                         lowLimit = dictionary;
                     } else {
-                        match = base + matchIndex;
+                        match = (const BYTE*) ((uptrval) base + matchIndex);
```

Comment:
we should rather ensure that, whenever this code is triggered, we have a guarantee that `base != NULL`.
`assert`ing this condition would be fine.